src/internal/test/fakes/fakes.go

```diff
@@ -6,27 +6,27 @@ import (
 	"github.com/futurehomeno/cliffhanger/storage"
 )
 
-type fakeConfigStorage struct {
+type fakeConfigStorage[T any] struct {
 	mu           sync.RWMutex
-	model        interface{}
-	modelFactory func() interface{}
+	model        T
+	modelFactory func() T
 }
 
-// NewConfigStorage returns a fake implementation for storage.Storage.
+// NewFakeStorage returns a fake implementation for storage.Storage.
 // Not suitable for production use.
-func NewConfigStorage(model interface{}, modelFactory func() interface{}) storage.Storage {
-	return &fakeConfigStorage{model: model, modelFactory: modelFactory}
+func NewFakeStorage[T any](model T, modelFactory func() T) storage.Storage[T] {
```

Comment:
I like the idea of making the storage generic, but I'd keep the previous name, so you don't repeat the word `fake` both in package and function names.